src/modules/olMap/formats/Mfp3Encoder.js

```diff
@@ -0,0 +1,599 @@
+		}
+
+	}
+
+	/**
+	 * Encodes the content of a OpenLayers {@see Map} to a MapFishPrint3 job request.
+	 * @param {ol.Map} olMap the map with the content to encode for MapFishPrint3
+	 * @returns {Object} the encoded mfp specs
+	 */
+	async encode(olMap) {
+
+		const getDefaultMapCenter = () => {
+			return new Point(olMap.getView().getCenter());
+		};
+		const getDefaultMapExtent = () => {
+			return olMap.getView().calculateExtent(olMap.getSize());
+		};
+
+		const mfpCenter = this._mfpProperties.mapCenter && typeof this._mfpProperties.mapCenter === Point
+			? this._mfpProperties.mapCenter.clone().transform(this._mapProjection, this._mfpProjection)
+			: getDefaultMapCenter().clone().transform(this._mapProjection, this._mfpProjection);
+
+		this._pageExtent = this._mfpProperties.pageExtent
+			? this._mfpProperties.pageExtent
+			: getDefaultMapExtent();
+
+		const shortLinkUrl = await this._generateShortUrl();
+		const qrCodeUrl = this._urlService.qrCode(shortLinkUrl);
+
+		const encodedLayers = olMap.getLayers().getArray()
+			.filter(layer => {
+				const layerExtent = layer.getExtent();
+
+				return layerExtent ? extentIntersects(layer.getExtent(), this._pageExtent) : true;
+			})
+			.flatMap(l => this._encode(l))
+			.reduce((layerSpecs, encodedLayer) => {
+				// todo: extract to method
+				const { attribution, thirdPartyAttribution, ...restSpec } = encodedLayer;
+				if (attribution || thirdPartyAttribution) {
+					const getLabels = (attribution) => Array.isArray(attribution?.copyright) ? attribution?.copyright?.map(c => c.label) : [attribution?.copyright?.label];
+
+					return {
+						specs: [...layerSpecs.specs, restSpec],
+						dataOwners: attribution ? [...layerSpecs.dataOwners, ...getLabels(attribution)] : [...layerSpecs.dataOwners],
+						thirdPartyDataOwners: thirdPartyAttribution ? [...layerSpecs.thirdPartyDataOwners, getLabels(thirdPartyAttribution)] : [...layerSpecs.thirdPartyDataOwners]
+					};
+				}
+				return { specs: [], dataOwners: [], thirdPartyDataOwners: [] };
+
+			}, { specs: [], dataOwners: [], thirdPartyDataOwners: [] });
+
+		const encodedOverlays = olMap.getOverlays().getArray().map(overlay => this._encodeOverlay(overlay));
+		return {
+			layout: this._mfpProperties.layoutId,
+			attributes: {
+				map: {
+					center: mfpCenter.getCoordinates(),
+					scale: this._mfpProperties.scale,
+					projection: this._mfpProjection,
+					dpi: this._mfpProperties.dpi,
+					rotation: this._mfpProperties.rotation,
+					layers: [...encodedLayers.specs, ...encodedOverlays]
+				},
+				dataOwner: encodedLayers.dataOwners.length !== 0 ? encodedLayers.dataOwners.join(',') : null,
+				thirdPartyDataOwner: encodedLayers.thirdPartyDataOwners.length !== 0 ? encodedLayers.thirdPartyDataOwners.join(',') : '',
+				shortLink: shortLinkUrl,
+				qrcodeurl: qrCodeUrl
+			}
+		};
+	}
+
+	_geoResourceFrom(layer) {
+		const layerId = layer.get('id');
+		const geoResourceId = layer.get('geoResourceId');
```

Comment:
presence of `geoResourceId` should be guaranteed

Reply:
Currently only VectorLayers have a `geoResourceId`. For WMTS- and WMS-Georesources the binding to a Layer is not done yet ([see LayerService.js  Line 83-97](https://github.com/ldbv-by/bav4-nomigration/blob/0e6e32ce157c1411c28d73a704324e8696d3c766/src/modules/olMap/services/LayerService.js#L89))

Reply:
presence of `geoResourceId` should be guaranteed

Reply:
Thx, your're right